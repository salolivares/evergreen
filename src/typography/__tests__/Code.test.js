import React from 'react'
import { render } from '@testing-library/react'
import faker from 'faker'
import Code from '../src/Code'

describe('Code', () => {
  it('should pass through `className` prop', () => {
    const expected = faker.random.word().toLowerCase()
    const component = (
      <Code data-testid="text" className={expected}>
        Testing
      </Code>
    )
    const { getByTestId } = render(component)
    expect(getByTestId('text')).toHaveClass(expected)
  })
})
