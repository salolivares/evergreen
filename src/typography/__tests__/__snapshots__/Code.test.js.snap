// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code <Code /> size 300 renders as expected 1`] = `
<code
  className="ub-fnt-sze_12px ub-f-wght_400 ub-ln-ht_16px ub-ltr-spc_0 ub-fnt-fam_tawojy ub-color_474d66 ub-bg-clr_rgba16-112-202-0-06 ub-bs_1o2cska ub-pl_6px ub-pr_6px ub-pb_3px ub-pt_3px ub-bblr_4px ub-bbrr_4px ub-btlr_4px ub-btrr_4px ub-box-szg_border-box"
>
  Text 300
</code>
`;

exports[`Code <Code /> size 300 with minimal appearance specified renders as expected 1`] = `
<code
  className="ub-fnt-sze_12px ub-f-wght_400 ub-ln-ht_16px ub-ltr-spc_0 ub-fnt-fam_tawojy ub-color_474d66 ub-box-szg_border-box"
>
  Text 300
</code>
`;

exports[`Code <Code /> size 400 renders as expected 1`] = `
<code
  className="ub-fnt-sze_14px ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_tawojy ub-color_474d66 ub-bg-clr_rgba16-112-202-0-06 ub-bs_1o2cska ub-pl_6px ub-pr_6px ub-pb_3px ub-pt_3px ub-bblr_4px ub-bbrr_4px ub-btlr_4px ub-btrr_4px ub-box-szg_border-box"
>
  Text 400
</code>
`;

exports[`Code <Code /> size 400 with minimal appearance specified renders as expected 1`] = `
<code
  className="ub-fnt-sze_14px ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_tawojy ub-color_474d66 ub-box-szg_border-box"
>
  Text 400
</code>
`;

exports[`Code <Code /> size 500 renders as expected 1`] = `
<code
  className="ub-fnt-sze_16px ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_tawojy ub-color_474d66 ub-bg-clr_rgba16-112-202-0-06 ub-bs_1o2cska ub-pl_6px ub-pr_6px ub-pb_3px ub-pt_3px ub-bblr_4px ub-bbrr_4px ub-btlr_4px ub-btrr_4px ub-box-szg_border-box"
>
  Text 500
</code>
`;

exports[`Code <Code /> size 500 with minimal appearance specified renders as expected 1`] = `
<code
  className="ub-fnt-sze_16px ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_tawojy ub-color_474d66 ub-box-szg_border-box"
>
  Text 500
</code>
`;

exports[`Code <Code /> size 600 renders as expected 1`] = `
<code
  className="ub-fnt-sze_18px ub-f-wght_400 ub-ln-ht_24px ub-ltr-spc_-0-07px ub-fnt-fam_tawojy ub-color_474d66 ub-bg-clr_rgba16-112-202-0-06 ub-bs_1o2cska ub-pl_6px ub-pr_6px ub-pb_3px ub-pt_3px ub-bblr_4px ub-bbrr_4px ub-btlr_4px ub-btrr_4px ub-box-szg_border-box"
>
  Text 600
</code>
`;

exports[`Code <Code /> size 600 with minimal appearance specified renders as expected 1`] = `
<code
  className="ub-fnt-sze_18px ub-f-wght_400 ub-ln-ht_24px ub-ltr-spc_-0-07px ub-fnt-fam_tawojy ub-color_474d66 ub-box-szg_border-box"
>
  Text 600
</code>
`;
